getting_started/guides/godot_essentials_guide.rst

```diff
@@ -0,0 +1,602 @@
+  tree systematically deletes itself from the bottom up!
+
+  Nodes in particular also have a
+  :ref:`queue_free() <class_Node_method_queue_free>` method. This delays
+  their deletion until the next idle frame to reduce any discernible
+  lag.
+
+Resources
+  Resources extend the :ref:`Reference <class_Reference>` class.
+  References are Objects that use "reference-counted" memory. These keep track
+  of how many references to the instance exist in 
+  `scope <https://en.wikipedia.org/wiki/Scope_(computer_science)>`__. Only when
+  all references are gone do they finally delete their instance.
+
+  For example, let's say you create an Object inside of a function and then end
+  the function. The Object's variable exits scope, removing all references to
+  the Object instance. But, you have not deleted the Object, so it's memory
+  continues to be in use. It has created a "memory leak." Every time you
+  execute that function, you leak more memory. If you execute this function
+  often, then it will consume ever-larger amounts of RAM. After a while, this
+  can lag the user's computer or even crash the program.
+
+  But, if you create a Reference inside of a function and then end
+  the function. When the variable exits scope, the Reference sees that the
+  last data reference is gone. It triggers an automatic ``free()`` on the
+  instance in the background. You do not leak any memory and your game continues
+  running smoothly.
+
+  Resources will track references in the same way. Loading a file for the
+  first time allocates a new instance. Later loads only return
+  the existing instance. If you no longer have any references, they
+  delete themselves for you.
+
+  .. note::
+  
+    The Godot Editor has a built-in feature to cache resources.
+
+Scripts and Scenes
+  Scripts are also resources. Every script file loads as a
+  :ref:`Script <class_Script>` instance. And because every script file is a
+  class, each constant that file declares is *static*. The data belongs to
+  the Script instance, not Object instances to which you've assigned it.
+
+  Godot likewise loads a scene file
+  into a :ref:`PackedScene <class_PackedScene>` instance. These are resources
+  that cache information about creating and initializing a node hierarchy.
+  You can even build your own scenes using the
+  :ref:`PackedScene.pack() <class_PackedScene_method_pack>` method.
+
+  .. note::
+
+    Godot's resources can both reference other resources and internally
+    store their own sub-resources.
+
+    This extends to scenes. They can have embedded resources such as
+    meshes, materials, shaders, particle effects, etc.
+    
+    However, Godot will try to save on memory allocations unless directed
+    otherwise. If you duplicate a scene, or if you copy/paste a
+    sub-resource, then it will NOT create a new resource. Instead, it
+    makes both scenes reference the same sub-resource.
+
+    To force Godot to create a new sub-resource, do X.
```

Comment:
Oh whoops. Looks like I left a todo in here.

Reply:
Iirc, you use the "Local to scene" checkbox to mark that a resource should create a new instance when the scene itself is copied. Then there's also a "Make Unique" operation that forces the current instance to *become* local to the current scene if it isn't already. Need to double check this info though.